src/main/java/com/fasterxml/jackson/core/io/IOContext.java

```diff
@@ -458,4 +479,12 @@ private IllegalArgumentException wrongBuf() {
         // sanity check failed; trying to return different, smaller buffer.
         return new IllegalArgumentException("Trying to release buffer smaller than original");
     }
+
+    @Override
+    public void close() {
+        if (!_closed) {
+            _bufferRecycler.release();
```

Comment:
Can we agree on a lifecycle? There is an argument to only release the recycler when the JsonFactory is closed. Meaning IOContext would not need a close method.

Reply:
Sorry, but I'm not understanding your comment on this. The `JsonFactory` is never closed afaik, maybe you mean the `JsonGenerator`?

Reply:
So, the close could be on the JsonFactory though. Or on the JsonGenerator. With IOContext, that makes 3 possible homes for the code that releases the buffer recycler. We should agree on which of the 3 homes makes the most sense.

Reply:
The only owner of the `BufferRecycler` (I mean the only holding a reference to it) and then the only one being responsible to release it back to the pool is the `IOContext`.

Reply:
I think close makes sense here; closing discussion.